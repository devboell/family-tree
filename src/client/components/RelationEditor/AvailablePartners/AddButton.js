import styled from 'styled-components'

const AddButton = styled.button`
  padding: 5px;
  margin-right: 5px;
  cursor: pointer;
`

export default AddButton
