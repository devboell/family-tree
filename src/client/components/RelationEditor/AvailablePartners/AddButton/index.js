import styled from 'styled-components'

const AddButton = styled.button`
  padding: 3px;
  margin-right: 5px;
`

export default AddButton
