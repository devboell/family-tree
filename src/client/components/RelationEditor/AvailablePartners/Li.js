import styled from 'styled-components'


const Li = styled.li`
  list-style: none;
  padding: 10px 10px;
  background-color: dark-gray;
  color:
`

export default Li
