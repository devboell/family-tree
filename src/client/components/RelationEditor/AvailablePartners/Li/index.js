import styled from 'styled-components'


const Li = styled.li`
  list-style: none;
  margin: 5px 0px;
  padding: 10px 10px;
  background-color: rgb(13, 9, 184);
`

export default Li
