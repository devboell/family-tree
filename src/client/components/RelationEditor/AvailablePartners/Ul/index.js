import styled from 'styled-components'


const Ul = styled.ul`
  flex-basis: 50%;
  list-style: none;
  height: 400px;
  margin-top: 20px;
  overflow-y: auto;
`

export default Ul
