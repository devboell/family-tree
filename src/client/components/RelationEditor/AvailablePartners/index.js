import React from 'react'
import { PropTypes } from 'prop-types'

import Ul from './Ul'
import Li from './Li'

const newPartner = person => ({
  relationId: null,
  partner: {
    id: person.id,
    name: person.name,
  },
  children: [],
})

const AvailablePartners = ({ persons, fields }) =>
  <Ul>
    {persons.map(person => (
      <Li key={`avail-partners-${person.id}`}>
        <button type="button" onClick={() => { fields.push(newPartner(person)) }}>add</button>
        {person.name}
      </Li>
    ))}
  </Ul>

AvailablePartners.propTypes = {
  persons: PropTypes.array.isRequired,
  fields: PropTypes.object.isRequired,
}


export default AvailablePartners
