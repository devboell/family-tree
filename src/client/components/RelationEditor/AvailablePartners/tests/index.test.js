import React from 'react'
import AvailablePartners from '../index'

const props = {
  fields: { push: jest.fn() },
  persons: [
    {
      id: 1,
      name: 'John',
    },
    {
      id: 2,
      name: 'Mary',
    },
  ],
}

it('AvailablePartners, renders without crashing', () => {
  expect(shallow(<AvailablePartners {...props} />)).toMatchSnapshot()
})

it('adds person to fields when button is clicked', () => {
  const wrapper = mount(<AvailablePartners {...props} />)
  const button1 = wrapper.find('button').at(0)
  const expected = {
    relationId: null,
    partner: {
      id: 1,
      name: 'John',
    },
    children: [],
  }
  button1.simulate('click')
  expect(props.fields.push).toHaveBeenCalledWith(expected)
})
