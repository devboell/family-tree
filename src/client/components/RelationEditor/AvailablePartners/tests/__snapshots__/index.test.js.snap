// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailablePartners, renders without crashing 1`] = `
<styled.ul>
  <styled.li>
    <styled.button
      onClick={[Function]}
      type="button"
    >
      add
    </styled.button>
    John
  </styled.li>
  <styled.li>
    <styled.button
      onClick={[Function]}
      type="button"
    >
      add
    </styled.button>
    Mary
  </styled.li>
</styled.ul>
`;
